rcl/test/rcl/test_rcl.cpp

```diff
@@ -117,9 +117,9 @@ TEST_F(CLASSNAME(TestRCLFixture, RMW_IMPLEMENTATION), test_rcl_init_and_ok_and_s
   {
     FakeTestArgv test_args;
     rcl_allocator_t failing_allocator = rcl_get_default_allocator();
-    failing_allocator.allocate = &failing_malloc;
-    failing_allocator.deallocate = failing_free;
```

Comment:
what's the motivation for not having a failing `deallocate` anymore? and for only setting `failing_calloc` in some test files (accident?)?

Reply:
Setting failing_calloc only in some files is a bug, I'll double check for those, but free doesn't return anything so it never fails. I suppose it could abort with double free, but it's not something we can easily test for nor would it likely increase our test coverage as no code in the functions we're testing would try to handle that case.

Reply:
Some cases (like this one) don't need failing calloc.